### PR TITLE
[FIX] point_of_sale: anglosaxon price unit for invoice from pos

### PR DESCRIPTION
**Pull request by william-andre**
Task 2041208
Followp of https://github.com/odoo/odoo/pull/35335

There was missing the part when an invoice is created from the POS

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by sle-odoo** [comment]
Can you please explain the issue and how you solved it in the commit message? I don't  get it.

File: addons/point_of_sale/models/pos_order.py [hunk 1]
```diff
@@ -1227,3 +1228,17 @@ def _get_report_values(self, docids, data=None):
...
+
+class AccountMoveLine(models.Model):
+    _inherit = 'account.move.line'
+
+    def _stock_account_get_anglo_saxon_price_unit(self):
+        self.ensure_one()
+        if not self.product_id:
+            return self.price_unit
+        price_unit = super(AccountMoveLine, self)._stock_account_get_anglo_saxon_price_unit()
+        order = self.env['pos.order'].search([('account_move', '=', self.move_id.id)])
```
Comment (sle-odoo) [inline addons/point_of_sale/models/pos_order.py]: so why is this code needed now and not before? was it lost during a refactoring? can you link the commit who broke it?

File: addons/point_of_sale/tests/test_anglo_saxon.py [hunk 2]
```diff
@@ -116,7 +119,7 @@ def test_create_account_move_line(self):
         self.assertEqual(aml_output.credit, self.product.standard_price, "Cost of Good Sold entry missing or mismatching")
         self.assertEqual(aml_expense.debit, self.product.standard_price, "Cost of Good Sold entry missing or mismatching")
 
-    def test_2041208(self):
+    def setup_2041208(self):
```
Comment (sle-odoo) [inline addons/point_of_sale/tests/test_anglo_saxon.py]: rename def _prepare_pos_order
add a docstring
``` python
""" Set the cost method of `self.product` as FIFO. Receive 5@5 and 5@1 and
create a `pos.order` record selling 7 units @ 450.
"""
```

File: addons/point_of_sale/tests/test_anglo_saxon.py [hunk 3]
```diff
@@ -155,8 +158,30 @@ def test_2041208(self):
             'amount_tax': 0,
             'amount_paid': 0,
             'amount_return': 0,
+        }
+
+        self.pos_order_pos0 = self.PosOrder.create(pos_order_values)
+
+    def test_2041208_without_invoice(self):
```
Comment (sle-odoo) [inline addons/point_of_sale/tests/test_anglo_saxon.py]: rename test_fifo_valuation_no_invoice and the other one test_fifo_valuation_with_invoice

File: addons/point_of_sale/tests/test_anglo_saxon.py [hunk 4]
```diff
@@ -155,8 +158,30 @@ def test_2041208(self):
             'amount_tax': 0,
             'amount_paid': 0,
             'amount_return': 0,
+        }
+
+        self.pos_order_pos0 = self.PosOrder.create(pos_order_values)
```
Comment (sle-odoo) [inline addons/point_of_sale/tests/test_anglo_saxon.py]: i would return self.PosOrder.create(pos_order_values)

File: addons/point_of_sale/tests/test_anglo_saxon.py (hunk 3 above)
Comment (sle-odoo) [reply addons/point_of_sale/tests/test_anglo_saxon.py]: add a docstirng
"""Register a payment and validate a session after selling a fifo product"""

File: addons/point_of_sale/tests/test_anglo_saxon.py [hunk 5]
```diff
@@ -155,8 +158,30 @@ def test_2041208(self):
...
+
+    def test_2041208_without_invoice(self):
+        self.setup_2041208()
```
Comment (sle-odoo) [inline addons/point_of_sale/tests/test_anglo_saxon.py]: order = self. _prepare_pos_order

File: addons/point_of_sale/tests/test_anglo_saxon.py [hunk 6]
```diff
@@ -155,8 +158,30 @@ def test_2041208(self):
...
+        context_make_payment = {"active_ids": [self.pos_order_pos0.id], "active_id": self.pos_order_pos0.id}
+        self.pos_make_payment_0 = self.PosMakePayment.with_context(context_make_payment).create({
+            'amount': 7 * 450.0,
```
Comment (sle-odoo) [inline addons/point_of_sale/tests/test_anglo_saxon.py]: why do you store it on self? there's no need

File: addons/point_of_sale/tests/test_anglo_saxon.py [hunk 7]
```diff
@@ -155,8 +158,30 @@ def test_2041208(self):
...
         })
 
+        # I click on the validate button to register the payment.
```
Comment (sle-odoo) [inline addons/point_of_sale/tests/test_anglo_saxon.py]:
```
# register the payment
context_make_payment...
payment = ...

# validate the session
current_session_id.action_pos_session_validate()

# check the anglo saxon move lines
self.assertEqual(line.debit...

File: addons/point_of_sale/tests/test_anglo_saxon.py [hunk 8]
```diff
@@ -155,8 +158,30 @@ def test_2041208(self):
...
+        context_payment = {'active_id': self.pos_order_pos0.id}
+        self.pos_make_payment_0.with_context(context_payment).check()
+        current_session_id = self.pos_config.current_session_id
+        current_session_id.action_pos_session_validate()
+
+        line = self.pos_order_pos0.account_move.line_ids.filtered(lambda l: l.debit and l.account_id == self.category.property_account_expense_categ_id)
+        self.assertEqual(self.pos_order_pos0.account_move.journal_id, self.pos_config.journal_id)
+        self.assertEqual(line.debit, 27, 'As it is a fifo product, the move\'s value should be 5*5 + 2*1')
+
+
```
Comment (sle-odoo) [inline addons/point_of_sale/tests/test_anglo_saxon.py]: extra new line

File: addons/point_of_sale/models/pos_order.py (hunk 1 above)
Comment (william-andre) [reply addons/point_of_sale/models/pos_order.py]: Is the commit message better now?  Everything else is done.

File: addons/point_of_sale/tests/test_anglo_saxon.py [hunk 9]
```diff
@@ -155,19 +161,51 @@ def test_2041208(self):
...
+        pos_order_pos0.account_move.post()
+
+        # validate the session
         current_session_id = self.pos_config.current_session_id
         current_session_id.action_pos_session_validate()
 
-        line = self.pos_order_pos0.account_move.line_ids.filtered(lambda l: l.debit and l.account_id == self.category.property_account_expense_categ_id)
+        # check the anglo saxon move lines
+        line = pos_order_pos0.account_move.line_ids.filtered(lambda l: l.debit and l.account_id == self.category.property_account_expense_categ_id)
+        self.assertEqual(pos_order_pos0.account_move.journal_id, self.sale_journal)
```
Comment (sle-odoo) [inline addons/point_of_sale/tests/test_anglo_saxon.py]: the diferences between the two tests are not obvious, is is just the journal_id the difference?

Comment (william-andre) [reply addons/point_of_sale/tests/test_anglo_saxon.py]: well, yes... But that difference is quite big as in this case it is the customer invoice, and in the other case it is the entry for the pos session. The lines should be in either one or the other depending on the invoicing policy for the pos.order.

File: addons/point_of_sale/models/pos_order.py (hunk 1 above)
Comment (sle-odoo) [reply addons/point_of_sale/models/pos_order.py]: maybe create the inverse one2many of pos.order account_move_id to access it more easily

Comment (sle-odoo) [reply addons/point_of_sale/models/pos_order.py]: and move all of this in a new file named account_move.py containing the inherit on account_move adding the o2m the inherit on account move line containing this _stock_account_get_anglo_saxon_price_unit

File: addons/point_of_sale/models/account_move.py [hunk 10]
```diff
@@ -0,0 +1,24 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
```
Comment (sle-odoo) [inline addons/point_of_sale/models/account_move.py]: api isn't used

File: addons/point_of_sale/models/pos_order.py (hunk 1 above)
Comment (william-andre) [reply addons/point_of_sale/models/pos_order.py]: done

File: addons/point_of_sale/models/account_move.py (hunk 10 above)
Comment (william-andre) [reply addons/point_of_sale/models/account_move.py]: ok

File: addons/point_of_sale/tests/test_anglo_saxon.py (hunk 9 above)
Comment (sle-odoo) [reply addons/point_of_sale/tests/test_anglo_saxon.py]: ok i finally understand the issue you're trying to fix
i'm not sure but shoudln't it be self. pos_config.invoice_journal_id ?

Comment (sle-odoo) [reply addons/point_of_sale/tests/test_anglo_saxon.py]: then again why did it stop working in master?

Comment (william-andre) [reply addons/point_of_sale/tests/test_anglo_saxon.py]: accountapocalypse, and yes but its the same thing.

Comment (sle-odoo) [reply addons/point_of_sale/tests/test_anglo_saxon.py]: > and yes but its the same thing 

yes but it's clearer for someone reading the test and comparing with the other test

Comment (sle-odoo) [reply addons/point_of_sale/tests/test_anglo_saxon.py]: > accountapocalypse 

was it really working before though? i quickly grepped around the v12 code and i don't see how it could use the invoice_journal_id for the anglo saxon entries .. but i don't really want to dig further so it's ok

**Comment by sle-odoo** [comment]
maybe reword the commit?

[FIX] point_of_sale: use the invoice journal of the config for the anglo saxon entries

File: addons/point_of_sale/tests/test_anglo_saxon.py (hunk 3 above)
Comment (sle-odoo) [reply addons/point_of_sale/tests/test_anglo_saxon.py]: dosctring not added

**Comment by sle-odoo** [comment]
[FIX] point_of_sale: anglo saxon price unit when invoicing before closing the session

**Comment by sle-odoo** [comment]
robodoo r+

thanks for the tests/cleanup and explanations
